tests/pyright/main.py

```diff
@@ -0,0 +1,129 @@
+# Slighly based on https://github.com/strawberry-graphql/strawberry/blob/main/tests/pyright/utils.py
+
+import os
+import subprocess
+import tempfile
+from dataclasses import dataclass
+from typing import List, cast
+
+from typing_extensions import Literal
+
+_cwd = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+_bin = os.path.join(_cwd, "node_modules", ".bin", "pyright")
+
+
+@dataclass
+class Result:
+    type: Literal["error", "info"]
+    message: str
+    line: int
+    column: int
+
+    @classmethod
+    def from_output_line(cls, output_line: str) -> "Result":
+        # an output line looks like: filename.py:11:6 - type: Message
+        file_info, result = output_line.split("-", maxsplit=1)
+
+        line, column = [int(value) for value in file_info.split(":")[1:]]
+        type_, message = [value.strip() for value in result.split(":", maxsplit=1)]
+        type_ = cast(Literal["error", "info"], type_)
+
+        return cls(type=type_, message=message, line=line, column=column)
+
+
+def run(code: str) -> List[Result]:
+    with tempfile.NamedTemporaryFile("w", suffix=".py") as f:
+        f.write(code)
+        f.flush()
+        result = subprocess.run([_bin, f.name], cwd=_cwd, stdout=subprocess.PIPE)
+
+    output = result.stdout.decode("utf-8")
+    results: List[Result] = []
+    for line in output.splitlines():
+        if line.strip().startswith(f"{f.name}:"):
+            results.append(Result.from_output_line(line))
+
+    return results
+
+
+_integer_test = run(
+    """\
+from django.db import models
+
+class Foo(models.Model):
+    integer_with_choices = models.IntegerField(
+        choices=[
+            (1, "First Option"),
+            (2, "Second Option"),
+        ],
+    )
+
+f = Foo()
+reveal_type(f.integer_with_choices)
+f.integer_with_choices == 2
+f.integer_with_choices == 3
+f.integer_with_choices = 3
+f.integer_with_choices = 2
+"""
+)
+assert _integer_test == [
+    Result(
+        type="info",
+        message='Type of "f.integer_with_choices" is "Literal[1, 2]"',
+        line=12,
+        column=13,
+    ),
+    Result(
+        type="error",
+        message='Condition will always evaluate to False since the types "Literal[1, 2]" and "Literal[3]" have no overlap (reportUnnecessaryComparison)',
+        line=14,
+        column=1,
+    ),
+    Result(
+        type="error",
+        message='Cannot assign member "integer_with_choices" for type "Foo" (reportGeneralTypeIssues)',
+        line=15,
+        column=3,
+    ),
+], _integer_test
+
+_char_test = run(
+    """\
+from django.db import models
+
+class Foo(models.Model):
+    char_with_choices = models.CharField(
+        choices=[
+            ("a", "A"),
+            ("b", "B"),
+        ],
+    )
+
+f = Foo()
+reveal_type(f.char_with_choices)
+f.char_with_choices == "c"
+f.char_with_choices == "a"
+f.char_with_choices = "c"
+f.char_with_choices = "b"
+"""
+)
+assert _char_test == [
```

Comment:
thoughts on wrapping this in pytest? might help with the error messages and usability

Reply:
Sure! Will do that right now!

I thought that maybe you might not want to add pytest as a dependency, but for sure it can help with error messages and usability.